incremental-font-transfer/src/ift_graph.rs

```diff
@@ -0,0 +1,261 @@
+        Tag::new(b"Harf"),
+        Tag::new(b"HARF"),
+        Tag::new(b"Buzz"),
+        Tag::new(b"BUZZ"),
+        //shapers
+
+        //arabic
+        Tag::new(b"init"),
+        Tag::new(b"medi"),
+        Tag::new(b"fina"),
+        Tag::new(b"isol"),
+        Tag::new(b"med2"),
+        Tag::new(b"fin2"),
+        Tag::new(b"fin3"),
+        Tag::new(b"cswh"),
+        Tag::new(b"mset"),
+        Tag::new(b"stch"),
+        //hangul
+        Tag::new(b"ljmo"),
+        Tag::new(b"vjmo"),
+        Tag::new(b"tjmo"),
+        //tibetan
+        Tag::new(b"abvs"),
+        Tag::new(b"blws"),
+        Tag::new(b"abvm"),
+        Tag::new(b"blwm"),
+        //indic
+        Tag::new(b"nukt"),
+        Tag::new(b"akhn"),
+        Tag::new(b"rphf"),
+        Tag::new(b"rkrf"),
+        Tag::new(b"pref"),
+        Tag::new(b"blwf"),
+        Tag::new(b"half"),
+        Tag::new(b"abvf"),
+        Tag::new(b"pstf"),
+        Tag::new(b"cfar"),
+        Tag::new(b"vatu"),
+        Tag::new(b"cjct"),
+        Tag::new(b"init"),
+        Tag::new(b"pres"),
+        Tag::new(b"abvs"),
+        Tag::new(b"blws"),
+        Tag::new(b"psts"),
+        Tag::new(b"haln"),
+        Tag::new(b"dist"),
+        Tag::new(b"abvm"),
+        Tag::new(b"blwm"),
+    ])
+}
+
+fn get_feature_tags(font: &FontRef<'_>) -> Result<BTreeSet<Tag>, ReadError> {
+    let standard_features = standard_features();
+
+    let mut result: BTreeSet<Tag> = Default::default();
+
+    if let Ok(gsub) = font.gsub() {
+        for fr in gsub.feature_list()?.feature_records() {
+            if standard_features.contains(&fr.feature_tag()) {
+                continue;
+            }
+
+            result.insert(fr.feature_tag());
+        }
+    }
+
+    if let Ok(gpos) = font.gpos() {
+        for fr in gpos.feature_list()?.feature_records() {
+            if standard_features.contains(&fr.feature_tag()) {
+                continue;
+            }
+
+            result.insert(fr.feature_tag());
+        }
+    }
+
+    Ok(result)
+}
+
+fn get_node_name(font: &FontRef<'_>) -> Result<String, ReadError> {
+    let chars: BTreeSet<char> = font
+        .charmap()
+        .mappings()
+        .map(|(cp, _)| char::from_u32(cp).unwrap())
+        .collect();
+
+    let mut name: String = chars.into_iter().collect();
+
+    let features = get_feature_tags(font)?;
+    if !features.is_empty() {
+        let features: Vec<_> = features.into_iter().map(|t| t.to_string()).collect();
+        let features = features.join(",");
+        name.push('|');
+        name.push_str(&features);
+    }
+
+    let axes = font.axes();
+    if !axes.is_empty() {
+        let axis_strings: Vec<_> = font
+            .axes()
+            .iter()
+            .map(|axis| {
+                axis.tag().to_string()
+                    + "["
+                    + &axis.min_value().to_string()
+                    + ".."
+                    + &axis.max_value().to_string()
+                    + "]"
```

Comment:
I... didn't even know there was an `Add` impl for String. That said, I think it would be more idiomatic (and presumably more efficient) to just use the `format!` macro here.